test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -117,7 +133,13 @@ func (sm *StackManager) GetStack(ctx context.Context, name string, config runner
 // GetStackNoDeleteOnFailure creates or return a stack based on stack name and config, if error occurs during stack creation, it will not destroy the created resources. Using this can lead to resource leaks.
 func (sm *StackManager) GetStackNoDeleteOnFailure(ctx context.Context, name string, config runner.ConfigMap, deployFunc pulumi.RunFunc, failOnMissing bool) (*auto.Stack, auto.UpResult, error) {
 
-	return sm.getStack(ctx, name, config, deployFunc, failOnMissing)
+	return sm.getStack(ctx, name, config, deployFunc, failOnMissing, nil)
+}
+
+// GetStackNoDeleteOnFailureWithLogger creates or return a stack based on stack name and config, if error occurs during stack creation, it will not destroy the created resources. Using this can lead to resource leaks.
+func (sm *StackManager) GetStackNoDeleteOnFailureWithLogger(ctx context.Context, name string, config runner.ConfigMap, deployFunc pulumi.RunFunc, failOnMissing bool, logWriter io.Writer) (*auto.Stack, auto.UpResult, error) {
```

Comment:
I'm not sure pilling up APIs is the right way forward. Given that's it's a straightforward change, can we just change to add the parameter?

Reply:
Deleted the extra method and added a parameter to the existing one